discovery/uyuni/uyuni.go

```diff
@@ -206,135 +192,107 @@ func NewDiscovery(conf *SDConfig, logger log.Logger) *Discovery {
-
-				// Check if system is monitoring entitled
-				for _, v := range details.Entitlements {
-					if v == "monitoring_entitled" { // golang has no native method to check if an element is part of a slice
-
-						// Get network details
-						netInfo, err = getSystemNetworkInfo(rpcclient, token, client.ID)
-						if err != nil {
-							level.Error(d.logger).Log("msg", "getSystemNetworkInfo failed", "clientId", client.ID, "err", err)
-							return
-						}
-
-						// Get list of groups this system is assigned to
-						candidateGroups, err := listSystemGroups(rpcclient, token, client.ID)
-						if err != nil {
-							level.Error(d.logger).Log("msg", "listSystemGroups failed", "clientId", client.ID, "err", err)
-							return
-						}
-						groups := []string{}
-						for _, g := range candidateGroups {
-							// get list of group formulas
-							// TODO: Put the resulting data on a map so that we do not have to repeat the call below for every system
-							if g.Subscribed == 1 {
-								groupFormulas, err := getGroupFormulaData(rpcclient, token, g.ID, "prometheus-exporters")
-								if err != nil {
-									level.Error(d.logger).Log("msg", "getGroupFormulaData failed", "groupId", client.ID, "err", err)
-									return
-								}
-								formulas = getCombinedFormula(formulas, groupFormulas)
-								// replace spaces with dashes on all group names
-								groups = append(groups, strings.ToLower(strings.ReplaceAll(g.SystemGroupName, " ", "-")))
-							}
-						}
-
-						// Get system formula list
-						systemFormulas, err := getSystemFormulaData(rpcclient, token, client.ID, "prometheus-exporters")
-						if err != nil {
-							level.Error(d.logger).Log("msg", "getSystemFormulaData failed", "clientId", client.ID, "err", err)
-							return
-						}
-						formulas = getCombinedFormula(formulas, systemFormulas)
-
-						// Iterate list of formulas and check for enabled exporters
-						for k, v := range formulas {
-							if v.Enabled {
-								port, err := extractPortFromFormulaData(v.Args)
-								if err != nil {
-									level.Error(d.logger).Log("msg", "Invalid exporter port", "clientId", client.ID, "err", err)
-									return
-								}
-								targets := model.LabelSet{}
-								addr := fmt.Sprintf("%s:%s", netInfo.IP, port)
-								targets[model.AddressLabel] = model.LabelValue(addr)
-								targets["exporter"] = model.LabelValue(k)
-								targets["hostname"] = model.LabelValue(details.Hostname)
-								targets["groups"] = model.LabelValue(strings.Join(groups, ","))
-								for _, g := range groups {
-									gname := fmt.Sprintf("grp_%s", g)
-									targets[model.LabelName(gname)] = model.LabelValue("active")
-								}
-								tg.Targets = append(tg.Targets, targets)
-							}
-						}
-					}
-				}
-				// Log debug information
-				if netInfo.IP != "" {
-					level.Info(d.logger).Log("msg", "Found monitored system", "Host", details.Hostname,
```

Comment:
I have noticed this can generate a lot of output if we have many monitored systems. Can you maybe change the level to 'debug'?